modding_tools/CMakeLists.txt

```diff
@@ -1,5 +1,9 @@
 add_executable(UserProfileTool user_profile_tool.cpp)
+# target_link_libraries(UserProfileTool PRIVATE
+#     SDL2::Main
+#     rigel_core
+# )
 target_link_libraries(UserProfileTool PRIVATE
-    SDL2::Main
```

Comment:
It should be possible to define "dummy" targets for `SDL2` when building with Emscripten, that way, we would not have to change these. I'll look into it!

Reply:
can you elaborate on that ? What do you mean by dummy targets ? My cmake-fu is still not strong enough :)

Reply:
Sure! So every time you say `add_library` or `add_executable` in CMake, you define a "target". Most of the time, a target results in code being compiled and a executable or library being produced. For example when we say `add_library(rigel_core`, it tells CMake to generate build files (Make files, VS solution etc.) that will compile all the sources listed for `rigel_core` and then link them into a static library.

Targets can depend on other targets using `target_link_libraries`. We use this to make the `RigelEngine` executable depend on the `rigel_core` library.

But targets are a more general concept, they don't always have to involve compiling code. A target can also be used to specify certain compiler flags, include paths, linker flags etc. Any target that depends on another target will then inherit all these compiler flags etc.

This is used in CMake for external dependencies. For example, we use `target_link_libraries` to specify that the target `rigel_core` should depend on the target `SDL2::Core`. The latter is a target  as well, just like `rigel_core`, but it does not cause any code to be compiled. Instead, it only defines the necessary include paths and compiler flags that need to be used in order to use the external SDL2 library. By saying that `rigel_core` depends on `SDL2::Core`, CMake knows that it needs to add all the compiler flags defined by `SDL2::Core` to any compiler command invocation that happens when building the files that are part of `rigel_core`.

Now where does the target `SDL2::Core` come from? It's defined by the find script, which we invoke when we use the `find_package(SDL2)` command. You can see how it's defining the target here: https://github.com/lethal-guitar/RigelEngine/blob/fbe161346b90ca8a43b74845d27a256a65fac0e7/cmake/Modules/FindSDL2.cmake#L354

Now my idea was that for the Emscripten case, we can define a target named `SDL2::Core` ourselves, and that target would specify that the `USE_SDL=2` flag should be passed to the compiler. Then we don't need to change any of the CMake code that uses the `SDL2::Core` target.

It would probably look something like this (not tested):

```cmake
add_library(SDL2::Core UNKNOWN IMPORTED)
set_target_properties(SDL2::Core PROPERTIES
  INTERFACE_COMPILE_OPTIONS "-s USE_SDL=2")
```
We could then do the same for SDL2_mixer and Boost.

Let me know if that helps!